### PR TITLE
namu.wiki - Ad ( PowerLink )

### PR DESCRIPTION
**Pull request by piquark6046**
https://github.com/AdguardTeam/AdguardFilters/issues/108883#issuecomment-1077053170
https://github.com/List-KR/List-KR/issues/236

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: filters-share/extended_css_ELEMHIDE.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+namu.wiki#?#article div:has( div.namuwiki-toc-ad ) ~ div:has( div.wiki-paragraph ):has(table > tbody > tr > td[colspan] > div > img[src*="w.namu.la/"]):not(:has( div.namuwiki-toc-ad ))
```
Comment (seia-soto) [inline filters-share/extended_css_ELEMHIDE.txt]: It's reasonable to remove the part: `div:has( div.wiki-paragraph )` for performance as almost every div in namuwiki has empty `.wiki-paragraph` node.

Comment (piquark6046) [reply filters-share/extended_css_ELEMHIDE.txt]: Cloud you give me a suggestion?

Comment (piquark6046) [reply filters-share/extended_css_ELEMHIDE.txt]: Did you mean this? @seia-soto 
```suggestion
namu.wiki#?#article div ~ div:has( div.wiki-paragraph ):has(table > tbody > tr > td[colspan] > div > img[src*="w.namu.la/"]):not(:has( div.namuwiki-toc-ad ))
```

Comment (seia-soto) [reply filters-share/extended_css_ELEMHIDE.txt]: Ah not that one. `.namuwiki-toc-ad` is required. I am building one. `div.wiki-paragraph` is meaningless as they update the site structure.

Comment (seia-soto) [reply filters-share/extended_css_ELEMHIDE.txt]: Based on my rule: `namu.wiki#?#article div:has(a[href*="/w/%EB%B6%84%EB%A5%98:"]) ~ div:has(td img[src^="//ww.namu.la/s/"][class]):not(:has(div.namuwiki-toc-ad, a[href^="#s-"]))`

You may change the rule as possible.

Comment (seia-soto) [reply filters-share/extended_css_ELEMHIDE.txt]: @piquark6046 I need cross-check if the rule is working for both side. I don't know how many versions that namuwiki is distributing are there so far but yours isn't working on my side.

`article div:has( div.namuwiki-toc-ad ) ~ div:has( div.wiki-paragraph ):has(table > tbody > tr > td[colspan] > div > img[src*="w.namu.la/"]):not(:has( div.namuwiki-toc-ad ))`

**Review by green1052 — APPROVED** [review]
lgtm

File: filters-share/extended_css_ELEMHIDE.txt (hunk 1 above)
Comment (piquark6046) [reply filters-share/extended_css_ELEMHIDE.txt]: https://github.com/List-KR/List-KR/pull/261#discussion_r834090968
This rule does not work on my site.

I will use this rule:
`namu.wiki#?#article div:has(a[href*="/w/%EB%B6%84%EB%A5%98:"]) ~ div:has(td img[src^="//ww.namu.la/s/"][class]):not(:has(div.namuwiki-toc-ad, a[href^="#s-"]))`

Comment (seia-soto) [reply filters-share/extended_css_ELEMHIDE.txt]: Confirmed.

**Comment by seia-soto** [comment]
@piquark6046 Update required for invalid `:has` pseudo use for minor browsers.

Choose one of following rule:

- Addition

```css
##article div:has(a[href*="/w/%EB%B6%84%EB%A5%98:"]) ~ div:has(td img[src^="//ww.namu.la/s/"][class]):not(:has(div.namuwiki-toc-ad):has(a[href^="#s-"])) table
```

- Deletion

```css
##article div:has(a[href*="/w/%EB%B6%84%EB%A5%98:"]) ~ div:has(td img[src^="//ww.namu.la/s/"][class]):not(:has(div.namuwiki-toc-ad)) table
```

**Comment by piquark6046** [comment]
@seia-soto Can you create a new PR?

**Pull request merged**